disnake/components.py

```diff
@@ -307,18 +330,151 @@ class SelectMenu(BaseSelectMenu):
 
     __repr_info__: ClassVar[Tuple[str, ...]] = BaseSelectMenu.__repr_info__ + __slots__
 
-    def __init__(self, data: SelectMenuPayload):
+    def __init__(self, data: StringSelectMenuPayload):
         super().__init__(data)
         self.options: List[SelectOption] = [
             SelectOption.from_dict(option) for option in data.get("options", [])
         ]
 
-    def to_dict(self) -> SelectMenuPayload:
-        payload = cast("SelectMenuPayload", super().to_dict())
+    def to_dict(self) -> StringSelectMenuPayload:
+        payload = cast("StringSelectMenuPayload", super().to_dict())
         payload["options"] = [op.to_dict() for op in self.options]
         return payload
 
 
+SelectMenu = StringSelectMenu  # backwards compatibility
+
+
+class UserSelectMenu(BaseSelectMenu):
+    """Represents a user select menu from the Discord Bot UI Kit.
+
+    .. versionadded:: 2.7
```

Comment:
```suggestion
    """Represents a user select menu from the Discord Bot UI Kit.

    .. note::
        The user constructible and usable type to create a
        user select menu is :class:`disnake.ui.UserSelect`.
        
    .. versionadded:: 2.7
```

Reply:
repeat for the other three classes

Reply:
https://github.com/DisnakeDev/disnake/pull/803/commits/9d653c1502344c03a31a922798d2a696c8dbcdea